sonar-platform-plugin/src/main/java/io/ballerina/sonar/SonarPlatformPlugin.java

```diff
@@ -0,0 +1,165 @@
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.sonar.Constants.END_LINE;
+import static io.ballerina.sonar.Constants.END_LINE_OFFSET;
+import static io.ballerina.sonar.Constants.FILE_NAME;
+import static io.ballerina.sonar.Constants.FILE_PATH;
+import static io.ballerina.sonar.Constants.FORWARD_SLASH;
+import static io.ballerina.sonar.Constants.ISSUE_FILE_PATH;
+import static io.ballerina.sonar.Constants.MESSAGE;
+import static io.ballerina.sonar.Constants.RULE_ID;
+import static io.ballerina.sonar.Constants.RULE_KIND;
+import static io.ballerina.sonar.Constants.RULE_PREFIX;
+import static io.ballerina.sonar.Constants.SOURCE;
+import static io.ballerina.sonar.Constants.START_LINE;
+import static io.ballerina.sonar.Constants.PLATFORM_NAME;
+import static io.ballerina.sonar.Constants.START_LINE_OFFSET;
+
+/**
+ * Represents the implementation of the {@link StaticCodeAnalysisPlatformPlugin} for reporting issues to SonarQube.
+ *
+ * @since 0.1.0
+ */
+public class SonarPlatformPlugin implements StaticCodeAnalysisPlatformPlugin {
+    private PlatformPluginContext platformPluginContext;
+    private final List<String> processBuilderArguments = new ArrayList<>();
+    private final ProcessBuilder processBuilder = new ProcessBuilder();
+    private final PrintStream outputStream = System.out;
+
+    @Override
+    public String platform() {
+        return PLATFORM_NAME;
+    }
+
+    @Override
+    public void init(PlatformPluginContext platformPluginContext) {
+        this.platformPluginContext = platformPluginContext;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            processBuilderArguments.add("cmd");
+            processBuilderArguments.add("/c");
+        } else {
+            processBuilderArguments.add("sh");
+            processBuilderArguments.add("-c");
+        }
+        processBuilderArguments.add("sonar-scanner");
+        processBuilderArguments.add("-Dsonar.exclusions=" +
+                "'" +
+                "**/*.java," +
+                "**/*.xml," +
+                "**/*.yaml," +
+                "**/*.go," +
+                "**/*.kt," +
+                "**/*.js," +
+                "**/*.html," +
+                "**/*.YAML" +
+                ",**/*.rb," +
+                "**/*.scala," +
+                "**/*.py" +
+                "'");
+    }
+
+    @Override
+    public void onScan(List<Issue> issues) {
+        saveIssues(ISSUE_FILE_PATH, issues);
+        if (platformPluginContext.initiatedByPlatform()) {
+            return;
+        }
+
+        processBuilderArguments.add("-DanalyzedResultsPath=" + Path.of(ISSUE_FILE_PATH).toAbsolutePath());
+        String sonarProjectPropertiesPath = platformPluginContext.platformArgs()
+                .getOrDefault("sonarProjectPropertiesPath", null);
```

Comment:
Do we need to explicitly give default value as `null` here? Just calling `get` should get you `null` if the key is abscent right?

Reply:
Yes, this is correct.